walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,48 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> availableTimeRange = new ArrayList<TimeRange>();
+        
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return availableTimeRange;
+        }
+        availableTimeRange.add(TimeRange.WHOLE_DAY);
+
+        Collection<String> meetingAttendees = request.getAttendees();
+        for (Event e : events) {
+            Collection<String> eventAttendees = e.getAttendees();
+            if (hasCommonAttendee (meetingAttendees, eventAttendees)) {
+                TimeRange eventTime = e.getWhen();
+                for (TimeRange t: new ArrayList<TimeRange>(availableTimeRange)) {
```

Comment:
Maybe we can extract the for-loop into separate method to make it easier to comprehend